File: online_judges/codechef/MODEQ.cpp
#include <bits/stdc++.h>

using namespace std;
typedef long long ll;

int MAX = 500001;
vector<vector<int> > divs(MAX, vector<int>() );

void init() {
    for(int i=1; i< MAX; i++)
        divs[i].push_back(1);

    for(int i=2; i< MAX; i++) {
        for(int j=i;j<MAX;j+=i){
            divs[j].push_back(i);
        }
    }
}


int main(){

    int t, n, m;
    init();
    scanf("%d", &t);
    for (int i=0;i<t;i++) {
        scanf("%d %d", &n, &m);

        ll ans = 0;
        for (int b=1;b<=n;b++) {
            int r = m % b;
            int xb = m - r;
            if (xb == 0) {
                ans += b - 1;
                continue;
            }

            for(auto a: divs[xb]) {
                if (a < b && (m % a) % b == r % a) {
                    ans ++;
                }
            }
        }
        printf("%lld\n", ans);
    }
    return 0;
}

File: online_judges/codechef/MODEQ.py


class Solution:
    def __init__(self):
        N = 5 * 10 ** 5 + 1
        self.divs = [[1] for i in range(N)]
        for i in range(2, N):
            for j in range(i, N, i):
                self.divs[j].append(i)

    def solve(self, n, m):
        ans = 0
        for b in range(1, n+1):
            r = m % b
            xb = m - r
            if xb == 0:
                ans += b - 1
                continue
            divs = self.divs[xb]
            for a in divs:
                if a < b and (m % a) % b == r % a:
                    ans += 1

        return ans


sol = Solution()
for _ in range(int(input())):
    n, m = [int(x) for x in input().split()]
    print(sol.solve(n, m))
